inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.cpp

```diff
@@ -170,57 +167,85 @@ void MKLDNNIfNode::initSupportedPrimitiveDescriptors() {
     }
+}
 
-    for (auto& map_rule : thenOutputPortMap) {
-        auto &toMem = getChildEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
-        auto &fromMem = outputMemThen[map_rule.to];
+void MKLDNNIfNode::prepareBeforeMappers(const bool is_then, const dnnl::engine& eng) {
+    auto &inputPortMap = is_then ? thenInputPortMap : elseInputPortMap;
+    auto &inputMems = is_then ? inputMemThen : inputMemElse;
+    auto &beforeMappers = is_then ? beforeThenMappers : beforeElseMappers;
+    for (auto& map_rule : inputPortMap) {
+        auto &fromMem = getParentEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
+        auto &toMem = inputMems[map_rule.to];
 
-        afterThenMappers.emplace_back(std::make_shared<PortMapHelper>(fromMem, toMem, eng));
+        beforeMappers.emplace_back(std::make_shared<PortMapHelper>(fromMem, toMem, eng));
     }
+}
 
-    for (auto& map_rule : elseOutputPortMap) {
+void MKLDNNIfNode::prepareAfterMappers(const bool is_then, const dnnl::engine& eng) {
+    auto &outputPortMap = is_then ? thenOutputPortMap : elseOutputPortMap;
+    auto &outputMems = is_then ? outputMemThen : outputMemElse;
+    auto &afterMappers = is_then ? afterThenMappers : afterElseMappers;
+    for (auto& map_rule : outputPortMap) {
         auto &toMem = getChildEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
-        auto &fromMem = outputMemElse[map_rule.to];
+        auto &fromMem = outputMems[map_rule.to];
 
-        afterElseMappers.emplace_back(std::make_shared<PortMapHelper>(fromMem, toMem, eng));
+        afterMappers.emplace_back(std::make_shared<PortMapHelper>(fromMem, toMem, eng));
     }
 }
 
 void MKLDNNIfNode::execute(mkldnn::stream strm) {
-    const bool condition = *(reinterpret_cast<const bool*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr()));
-
-    if (condition) {
-        for (auto &mapper : beforeThenMappers)
-            mapper->execute(strm);
-        subGraphThen.ResetInferCount();
-        subGraphThen.Infer();
-        for (auto &mapper : afterThenMappers)
-            mapper->execute(strm);
-    } else {
-        for (auto &mapper : beforeElseMappers)
-            mapper->execute(strm);
-        subGraphElse.ResetInferCount();
-        subGraphElse.Infer();
-        for (auto &mapper : afterElseMappers)
-            mapper->execute(strm);
-    }
+    condition = (reinterpret_cast<const bool*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr()))[0];
+    auto& beforeMappers = condition ? beforeThenMappers : beforeElseMappers;
+    auto& afterMappers = condition ? afterThenMappers : afterElseMappers;
+    auto& subGraph = condition ? subGraphThen : subGraphElse;
+
+    for (auto &mapper : beforeMappers)
+        mapper->execute(strm);
+    subGraph.ResetInferCount();
+    subGraph.Infer();
+    for (auto &mapper : afterMappers)
+        mapper->execute(strm);
+}
+
+void MKLDNNIfNode::executeDynamicImpl(mkldnn::stream strm) {
+    auto& beforeMappers = condition ? beforeThenMappers : beforeElseMappers;
```

Comment:
looks like code duplication with `execute`
maybe we can place all of this into `execute` and add check `isDynamicNode` to understand that we need to call `prepareAfterMappers`

Reply:
I am not sure about it because it is different implementations with the same one part of code... But for different modes :) I don't want to shuffle these implementations

Reply:
what about extracting 224 - 233 lines in separate method and call it in both functions?

Reply:
Okey, I gave up :)), one common `execute` would be better that separate method. Done